restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,374 @@
+          List<UserDefinedTypeResponse> udtResponses =
+              authenticatedDB.getTypes(keyspaceName).stream()
+                  .map(this::mapUdtAsReponse)
+                  .collect(Collectors.toList());
+          // Filter based on user credentials
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  udtResponses.stream()
+                      .map(UserDefinedTypeResponse::getName)
+                      .collect(Collectors.toList()),
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+          // Wrap response if user does not asked for raw.
+          Object response =
+              raw ? udtResponses : new ResponseWrapper<List<UserDefinedTypeResponse>>(udtResponses);
+          return Response.status(Response.Status.OK)
+              .entity(Converters.writeResponse(response))
+              .build();
+        });
+  }
+
+  @Timed
+  @GET
+  @ApiOperation(
+      value = "Get a user defined type (UDT) from its identifier",
+      notes = "Retrieve data for a single table in a specific keyspace.",
+      response = ResponseWrapper.class)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = Table.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 404, message = "Not Found", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  @Path("/{typeName}")
+  public Response findById(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(
+              value = "Name of the user defined type (UDT) to use for the request.",
+              required = true)
+          @PathParam("typeName")
+          final String typeName,
+      @ApiParam(value = "Unwrap results", defaultValue = "false") @QueryParam("raw")
+          final boolean raw,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          // Accessing datastore
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          // Permissions on a type are the same as keyspace (CreateTypeFetcher for graphQl API)
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  null,
```

Comment:
When retrieving all types there's a collection here. Would be the case of setting there as null or using a singletonList here?